### PR TITLE
feat: add project entity service

### PR DESCRIPTION
**Pull request by srdkr**
Resolves: #EPMLSTRCMW-115

### CODE REVIEW

File: portal/src/main/scala/cromwell/pipeline/datastorage/dao/repository/ProjectRepository.scala [hunk 1]
```diff
@@ -16,6 +16,7 @@ class ProjectRepository(pipelineDatabaseEngine: PipelineDatabaseEngine, projectE
 
   def addProject(project: Project): Future[ProjectId] = database.run(projectEntry.addProjectAction(project))
 
-  def deactivateById(projectId: ProjectId): Future[Int] = database.run(projectEntry.deactivateProjectById(projectId))
+  def deactivateProjectById(projectId: ProjectId): Future[Int] =
+    database.run(projectEntry.deactivateProjectById(projectId))
```
Comment (hope-doe) [inline portal/src/main/scala/cromwell/pipeline/datastorage/dao/repository/ProjectRepository.scala]: please use the same naming in all files (deactivateProjectById looks good, but UserRepository still provides deactivateById def)

File: portal/src/main/scala/cromwell/pipeline/datastorage/dto/Project.scala [hunk 2]
```diff
@@ -11,3 +11,5 @@ final case class Project(
 )
 
 final case class ProjectId(value: String) extends MappedTo[String]
+
+final case class AddProjectRequest(ownerId: UserId, name: String, repository: String)
```
Comment (hope-doe) [inline portal/src/main/scala/cromwell/pipeline/datastorage/dto/Project.scala]: case class names should not be verbs, ProjectAdditionRequest (or some other noun) is much more suitable

File: portal/src/main/scala/cromwell/pipeline/datastorage/dao/repository/ProjectRepository.scala (hunk 1 above)
Comment (srdkr) [reply portal/src/main/scala/cromwell/pipeline/datastorage/dao/repository/ProjectRepository.scala]: done

File: portal/src/main/scala/cromwell/pipeline/datastorage/dto/Project.scala (hunk 2 above)
Comment (srdkr) [reply portal/src/main/scala/cromwell/pipeline/datastorage/dto/Project.scala]: renamed

File: portal/src/test/scala/cromwell/pipeline/service/ProjectServiceTest.scala [hunk 3]
```diff
@@ -0,0 +1,73 @@
+package cromwell.pipeline.service
+
+import cromwell.pipeline.datastorage.dao.repository.ProjectRepository
+import cromwell.pipeline.datastorage.dto.{ Project, ProjectAdditionRequest, ProjectId, UserId }
+import org.scalamock.scalatest.MockFactory
```
Comment (ilyaonishenko) [inline portal/src/test/scala/cromwell/pipeline/service/ProjectServiceTest.scala]: We don't use scalamock anymore, we use `Mockito`.

File: portal/src/test/scala/cromwell/pipeline/service/ProjectServiceTest.scala [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{ Matchers, WordSpec }
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+class ProjectServiceTest extends WordSpec with Matchers with MockFactory with ScalaFutures {
```
Comment (ilyaonishenko) [inline portal/src/test/scala/cromwell/pipeline/service/ProjectServiceTest.scala]: Please use `AsyncWordSpec`

File: portal/src/test/scala/cromwell/pipeline/service/ProjectServiceTest.scala (hunk 3 above)
Comment (srdkr) [reply portal/src/test/scala/cromwell/pipeline/service/ProjectServiceTest.scala]: done

File: portal/src/test/scala/cromwell/pipeline/service/ProjectServiceTest.scala (hunk 4 above)
Comment (srdkr) [reply portal/src/test/scala/cromwell/pipeline/service/ProjectServiceTest.scala]: done

**Pull request merged**